### PR TITLE
Update classification section 2

### PR DESCRIPTION
**Pull request by YinYi000**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] [local build the book](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-book)
- [ ] review the artifacts from local build
- [ ] [local build the slides](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-slides-optional) if appliable
- [ ] Unit test if appliable

# Checklist:

- [x] My code follows the [style guidelines](https://github.com/open-academy/machine-learning/blob/main/STYLE_GUIDE.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have applied tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: open-machine-learning-jupyter-book/ml-fundamentals/classification/more-classifiers.md [hunk 1]
```diff
@@ -14,3 +14,221 @@ kernelspec:
...
+
+Essentially there are two important parameters - `multi_class` and `solver` - that we need to specify, when we ask Scikit-learn to perform a logistic regression. The `multi_class` value applies a certain behavior. The value of the solver is what algorithm to use. Not all solvers can be paired with all `multi_class` values.
+
+According to the docs, in the multiclass case, the training algorithm:
+
+- **Uses the one-vs-rest (OvR) scheme**, if the `multi_class` option is set to `ovr`
+- **Uses the cross-entropy loss**, if the `multi_class` option is set to `multinomial`. (Currently the `multinomial` option is supported only by the ‘lbfgs’, ‘sag’, ‘saga’ and ‘newton-cg’ solvers.)"
+
+```{seealso}
+The 'scheme' here can either be 'ovr' (one-vs-rest) or 'multinomial'. Since logistic regression is really designed to support binary classification, these schemes allow it to better handle multiclass classification tasks. [source](https://machinelearningmastery.com/one-vs-rest-and-one-vs-one-for-multi-class-classification/)
```
Comment (reveurmichael) [inline open-machine-learning-jupyter-book/ml-fundamentals/classification/more-classifiers.md:140]: for all source, we will use `&#x1F517;source` instead. An emoji is used.

File: open-machine-learning-jupyter-book/ml-fundamentals/classification/more-classifiers.md [hunk 2]
```diff
@@ -14,3 +14,221 @@ kernelspec:
...
+
+Scikit-learn offers this table to explain how solvers handle different challenges presented by different kinds of data structures:
+
+```{figure} ../../../images/ml-fundamentals/ml-classification/solvers.png
+---
+name: 'solvers'
+width: 90%
+---
+Image by [Jen Looper](https://twitter.com/jenlooper)
+```
```
Comment (reveurmichael) [inline open-machine-learning-jupyter-book/ml-fundamentals/classification/more-classifiers.md:153]: Let's use a real HTML or markdown table. How to generate a table: 
- https://www.tablesgenerator.com/html_tables
- https://www.tablesgenerator.com/markdown_tables

File: open-machine-learning-jupyter-book/ml-fundamentals/classification/more-classifiers.md [hunk 3]
```diff
@@ -14,3 +14,221 @@ kernelspec:
...
+### A better approach
+
+A better way than wildly guessing, however, is to follow the ideas on this downloadable [ML Cheat sheet](https://docs.microsoft.com/azure/machine-learning/algorithm-cheat-sheet?WT.mc_id=academic-77952-leestott). Here, we discover that, for our multiclass problem, we have some choices:
+
+```{figure} ../../../images/ml-fundamentals/ml-classification/cheatsheet.png
+---
+name: 'comparison of classifiers'
+width: 90%
+---
+Image by [Jen Looper](https://twitter.com/jenlooper)
```
Comment (reveurmichael) [inline open-machine-learning-jupyter-book/ml-fundamentals/classification/more-classifiers.md:108]: Caption should be more accurate. Not need to mention the auther. Add a source if needed.

File: open-machine-learning-jupyter-book/ml-fundamentals/classification/more-classifiers.md (hunk 2 above)
Comment (YinYi000) [reply open-machine-learning-jupyter-book/ml-fundamentals/classification/more-classifiers.md]: Here, the table is a picture.

**Review by reveurmichael — APPROVED** [review]
Everything OK now.

File: open-machine-learning-jupyter-book/assignments/ml-fundamentals/study-the-solvers.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Study the solvers
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/ml-fundamentals/study-the-solvers.md:1]: Could you please double check if this page could be linked correctly from another page? Where you are creating a link pointing to this, please check if that URL is correct.

File: open-machine-learning-jupyter-book/ml-fundamentals/classification/more-classifiers.md [hunk 5]
```diff
@@ -14,3 +14,221 @@ kernelspec:
...
 
 # More classifiers
+
+In this section, you will use the dataset you saved from the last section full of balanced, clean data all about cuisines.
+
+You will use this dataset with a variety of classifiers to _predict a given national cuisine based on a group of ingredients_. While doing so, you'll learn more about some of the ways that algorithms can be leveraged for classification tasks.
+
+## Exercise - predict a national cuisine
+
+1\. Working in this section's [build-classification-models](../../assignments/ml-fundamentals/build-classification-models.ipynb) folder, import that file along with the Pandas library:
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/ml-fundamentals/classification/more-classifiers.md:24]: this is not folder. you may rephrase it a little bit.

File: open-machine-learning-jupyter-book/ml-fundamentals/classification/more-classifiers.md [hunk 6]
```diff
@@ -14,3 +14,221 @@ kernelspec:
...
+
+```{code-cell}
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split, cross_val_score
+from sklearn.metrics import accuracy_score,precision_score,confusion_matrix,classification_report, precision_recall_curve
+from sklearn.svm import SVC
+import numpy as np
+```
+
+3\. Divide the X and y coordinates into two dataframes for training. `cuisine` can be the labels dataframe:
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/ml-fundamentals/classification/more-classifiers.md:44]: X should be x?

File: open-machine-learning-jupyter-book/assignments/ml-fundamentals/study-the-solvers.md (hunk 4 above)
Comment (YinYi000) [reply open-machine-learning-jupyter-book/assignments/ml-fundamentals/study-the-solvers.md:1]: yes, it is correct, I can open the link to this file. the link is in this file : "more-classifiers.md"

**Pull request merged**